### PR TITLE
#1072 implemented VbEvent

### PR DESCRIPTION
**Pull request by krzyk**
#1072 
* implemented VbEvent Simple and Message + tests

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1328](https://github.com/zerocracy/farm/issues/1328) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1328](https://github.com/zerocracy/farm/issues/1328) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1841930141); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/zerocracy/radars/viber/VbEventTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Test
+    public void parsesMessage() {
+        try (
+            final JsonReader reader = Json.createReader(
+                VbEventTest.class.getResourceAsStream("message.json")
+            )
+        ) {
+            final JsonObject json = reader.readObject();
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/radars/viber/VbEventTest.java]: @krzyk we can inline this

File: src/test/java/com/zerocracy/radars/viber/VbEventTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    @Test
+    public void parsesSimply() {
+        try (
+            final JsonReader reader = Json.createReader(
+                VbEventTest.class.getResourceAsStream("message.json")
+            )
+        ) {
+            final JsonObject json = reader.readObject();
+            final VbEvent event = new VbEvent.Simple(json);
+            MatcherAssert.assertThat(event.event(), Matchers.is("message"));
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/radars/viber/VbEventTest.java]: @krzyk lines 47-55 assertions are redundant with `parsesMessage`. Maybe just check the JSON here?

Comment (krzyk) [reply src/test/java/com/zerocracy/radars/viber/VbEventTest.java]: @carlosmiranda but I think we need to check that the delegation happens, right?

**Comment by krzyk** [comment]
@carlosmiranda updated, please take a look

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1328](https://github.com/zerocracy/farm/issues/1328) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1328#issuecomment-402615680))

@carlosmiranda This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1328](https://github.com/zerocracy/farm/issues/1328) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 19hr
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231328%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@carlosmiranda Thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**